monitoring/scd_qualifier/flight_volumes_generator.py

```diff
@@ -0,0 +1,307 @@
+        N_COLS = 3
+        N_ROWS = 2
+        cell_size_x = (self.maxx - self.minx)/(N_COLS)  # create three columns
+        cell_size_y = (self.maxy - self.miny)/(N_ROWS)  # create two rows
+        grid_cells = []
+        for u0 in range(0, N_COLS):  # 3 columns
+            x0 = self.minx + (u0 * cell_size_x)
+            for v0 in range(0, N_ROWS):  # 2 rows
+                y0 = self.miny + (v0 * cell_size_y)
+                x1 = x0 + cell_size_x
+                y1 = y0 + cell_size_y
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))
+        self.grid_cells = grid_cells
+
+    def utm_converter(self, shapely_shape: shapely.geometry, inverse:bool=False) -> shapely.geometry.shape:
+        ''' A helper function to convert from lat / lon to UTM coordinates for buffering. tracks. This is the UTM projection (https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system), we use Zone 33T which encompasses Switzerland, this zone has to be set for each locale / city. Adapted from https://gis.stackexchange.com/questions/325926/buffering-geometry-with-points-in-wgs84-using-shapely '''
+
+        proj = Proj(proj="utm", zone=self.utm_zone, ellps="WGS84", datum="WGS84")
+
+        geo_interface = shapely_shape.__geo_interface__
+        feature_type = geo_interface['type']
+        coordinates = geo_interface['coordinates']
+        if feature_type == 'Polygon':
+            new_coordinates = [[proj(*point, inverse=inverse) for point in linring] for linring in coordinates]
+        elif feature_type == 'LineString':
+            new_coordinates = [proj(*point, inverse=inverse) for point in coordinates]
+        else:
+            raise RuntimeError('Unexpected geo_interface type: {}'.format(feature_type))
+
+        return shapely.geometry.shape({'type': feature_type, 'coordinates': tuple(new_coordinates)})
+
+    def input_extents_valid(self) -> None:
+        ''' This method checks if the input extents are valid i.e. small enough, if the extent is too large, we reject them, at the moment it checks for extents less than 500m x 500m square but can be changed as necessary.'''
+
+        box = shapely.geometry.box(self.minx, self.miny, self.maxx, self.maxy)
+        area = abs(self.geod.geometry_area_perimeter(box)[0])
+
+        # Have a area less than 500m x 500m square and more than 300m x 300m square to ensure enough space for tracks
+        if (area) < 250000 and (area) > 90000:
+            return
+        else:
+            raise ValueError("The extents provided are not of the correct size, please provide extents that are less than 500m x 500m and more than 300m x 300m square")
+        
+    def generate_random_flight_path_polygon(self, generate_polygon:bool) -> Union[LineString, Polygon]:
```

Comment:
Why combine two entirely separate behaviors into one method with behavior controlled by a bool flag?  It seems like there should be two separate methods instead.

Reply:
Thank you for this, I have now split them in to two smaller methods.